### PR TITLE
Introduce Ruff as formatter

### PR DESCRIPTION
**Pull request by hirosassa**
## What I did in this PR
related to #344 #345 

I replaced `yapf` with `ruff` as formatter.

## discussion

Coding style of YAPF and Ruff is slightly different. We need to discuss on the style.
The big difference is as follows:
- remove a blank line after class [ref](https://github.com/hirosassa/gokart/compare/intro-ruff...hirosassa:gokart:replace-yapf?expand=1#diff-a2775bccf635af6a3c215299acfd9131206fc0c6e27a375a6d2e5e8ce17a37eeL9)
- insert a whitespace between index [ref](https://github.com/hirosassa/gokart/compare/intro-ruff...hirosassa:gokart:replace-yapf?expand=1#diff-d30054885311f8eb77523063db114b013d94dd98d39a2d4e34a6f3ec5f461bb1L73-L74)
- remove unnecessary new line operator [ref](https://github.com/hirosassa/gokart/compare/intro-ruff...hirosassa:gokart:replace-yapf?expand=1#diff-d30054885311f8eb77523063db114b013d94dd98d39a2d4e34a6f3ec5f461bb1L139-L141)
- change long arguments indentation style [ref](https://github.com/hirosassa/gokart/compare/intro-ruff...hirosassa:gokart:replace-yapf?expand=1#diff-953ea2208ceb72f318b511ed3dc73c7adae0fda58bd279a677641b2212f1d21aL63-L69)

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -13,15 +13,15 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/setup-python@v4
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
       with:
-        python-version: 3.11
+        python-version: 3.12
```
Comment (kitagry) [inline .github/workflows/format.yml:19]: :+1:

**Review by kitagry — APPROVED** [review]
LGTM

I love fewer dependencies! Thank you!

**Comment by hirosassa** [comment]
@Hi-king @yokomotod @mski-iksm pls check

**Comment by hirosassa** [comment]
Config is applied in [this commit](https://github.com/m3dev/gokart/pull/346/commits/195883a3371efb566f4ff3962bbe818ff5e09fe2).
Ruff formatter is applied in [this commit](https://github.com/m3dev/gokart/pull/346/commits/1312e6098562ceee5d862fb0c75c7e2f8e9393c0)

**Comment by Hi-king** [comment]
LGTM.  Let's make a bold change!

**Comment by yokomotod** [comment]
Sorry for late the late reply!

**Pull request merged**